### ISSUE TITLE
Issue with "not" in geojson schema

### ISSUE THREAD

**Issue by beschoenen** [justinrainbow/json-schema#566]
I'm trying to validate a simple object against the geojson schema but it fails on the "not" in the `crs` property.

This is the test I wrote, which currently fails.

```php
public function testValidateWithNonExistedPropertyUsingNot()
{
    $schema = json_decode(file_get_contents('http://json.schemastore.org/geojson'), true);
    $data = json_decode('{"type": "FeatureCollection", "features": []}', true);

    $validator = new Validator();
    $validator->validate($data, $schema);

    $this->assertTrue($validator->isValid());
}
```

It looks like it has something do to with "not" on properties that are not required/present in the data.

**Comment by erayd** [comment]
@beschoenen It looks like you're passing an associative array as both the schema and the data. Don't do this; it's incorrect - just pass it a normally decoded object and don't muck with the types. JSON has no native assoc type; an array is not a valid schema.

**Comment by erayd** [comment]
@beschoenen Despite the issue with your example I mentioned above regarding passing assoc arrays, it does look like there's something funky going on regarding that `crs` property. I'm investigating...

**Comment by erayd** [comment]
@beschoenen Thanks for reporting this - it should be fixed in #567. I'll backport that into the 5.x.x branch once it's merged.

The root cause of the bug was attempting to run some validation logic on undefined (but schema-constrained) properties, when that logic requires a defined instance in order to make sense, and should never have been running against an undefined value at all.

**Comment by beschoenen** [comment]
Thank you for taking the time to research this, greatly appreciated.
